src/librustc/ty/query/plumbing.rs

```diff
@@ -689,76 +701,86 @@ impl<'tcx> TyCtxt<'tcx> {
 
         // If `-Zincremental-verify-ich` is specified, re-hash results from
         // the cache and make sure that they have the expected fingerprint.
-        if unlikely!(self.sess.opts.debugging_opts.incremental_verify_ich) {
-            self.incremental_verify_ich::<Q>(&result, dep_node, dep_node_index);
+        if unlikely!(tcx.session().opts.debugging_opts.incremental_verify_ich) {
```

Comment:
Make `incremental_verify_ich` and `incremental_queries` methods on `QueryContext` to avoid the need for `session()`.